### DISCUSSION TITLE
RandomGeoBatchSampler with Time Constraints`

### DISCUSSION THREAD

**Discussion by TolgaAktas** [torchgeo/torchgeo#1591] [Q&A]
Greetings,

I was wondering whether it's possible to do the following without changing the torchgeo's underlying code. I am happy with RandomGeoBatchSampler providing a number of NxN bounding boxes however I would like to put mint, maxt constraints on the random sampling procedure. The idea is to sample images from the Landsat dataset which are taken at proximal dates so they have a higher similarity in terms of radiance and reflectance. Otherwise, there's a mismatch, let's say, if we sample the area but the pairs of images come from separate seasons thus they are not similar. 

How can we achieve this?

**Comment by adamjstewart** [comment]
I don't think this would be too difficult to support. Right now, the `size` passed into each sampler can be a single int (height/width) or a tuple of two ints (height, width). We could also add support for a tuple of three ints (height, width, time). You would need to mess with `_to_tuple` to convert all of the above to a tuple of length 3 (with time being [maxt - mint] if in the first 2 cases). Then you would modify `torchgeo.samplers.utils.get_random_bounding_box` to actually use the time dimension. Want to try to submit a PR?
